internal/listener/listener.go

```diff
@@ -86,13 +85,17 @@ func (l *Listener) ProcessEvent(w http.ResponseWriter, req *http.Request) {
 			msg = fmt.Sprintf(format, a...)
 		}
 
-		logger := l.logger.With(zap.Int("status-code", statusCode), zap.String("msg", msg))
+		logger := l.logger.With(zap.Int("status_code", statusCode), zap.String("message", msg))
 		if ev != nil {
 			logger = logger.With(zap.Stringer("event", ev))
 		}
 
+		if statusCode >= http.StatusInternalServerError {
+			msg = event.ErrEventProcessing.Error()
+		}
+
 		http.Error(w, msg, statusCode)
-		logger.Debugw("Abort listener submitted event processing")
+		logger.Error("Failed to successfully process event")
```

Comment:
I'd do this redaction of the error outside of this function. Then you don't have to rely on the status code here. Additionally, the way you do the logging now, this would log something like "POST required" at the error level which doesn't sound very useful to me, thus I'd also leave this at debug level here and instead add an error log for that specific error case where this function is called.